subm/src/request1bld.js

```diff
@@ -0,0 +1,220 @@
+/* eslint-disable no-await-in-loop */
+// See https://github.com/Agoric/validator-profiles/wiki/Request-1-BLD
+
+const { DiscordAPI, getContent, paged } = require('./discordGuild.js');
+const { upsert } = require('./sheetAccess');
+const { searchBySender, transfers } = require('./tendermintRPC');
+
+const config = {
+  host: 'rpc-agoric.nodes.guru',
+  address: 'agoric15qxmfufeyj4zm9zwnsczp72elxsjsvd0vm4q8h',
+  quorum: 2,
+};
+
+const fail = () => {
+  throw Error();
+};
+
+/**
+ * @param {ReturnType<ReturnType<typeof DiscordAPI>['channels']>} channel
+ * @param {ReturnType<ReturnType<typeof DiscordAPI>['guilds']>} guild
+ * @param {Snowflake} role
+ * @param {number} quorum
+ * @yields {{ message: Message, endorsers: User[] }}
+ * @typedef {import('./discordGuild').Snowflake} Snowflake
+ */
+async function* authorizedRequests(channel, guild, role, quorum) {
+  /** @type {Map<Snowflake, import('./discordGuild').GuildMember>} */
+  const memberDetail = new Map();
+  /** @param {Snowflake} id */
+  const getMemberDetail = async id => {
+    if (memberDetail.has(id)) {
+      return memberDetail.get(id) || fail();
+    }
+    const detail = await guild.members(id);
+    // console.log(detail);
+    memberDetail.set(id, detail);
+    return detail;
+  };
+
+  const messages = await paged(channel.getMessages);
+  const hasAddr = messages.filter(msg => msg.content.match(/agoric1/));
+  if (!hasAddr) return;
+  const hasChecks = hasAddr.filter(msg => {
+    const [checks] = (msg.reactions || []).filter(r => r.emoji.name === '✅');
+    return (checks || {}).count >= quorum;
+  });
+  if (!hasChecks) return;
+
+  for (const msg of hasChecks) {
+    const endorsements = await channel.messages(msg.id).reactions('✅');
+    const endorsers = [];
+    for (const endorsement of endorsements) {
+      const detail = await getMemberDetail(endorsement.id);
+      if (detail && detail.roles && detail.roles.includes(role)) {
+        endorsers.push(detail);
+      }
+    }
+    if (endorsers.length >= quorum) {
+      const [_, address] = msg.content.match(/(agoric1\S+)/);
```

Comment:
need tighter regex match + bech32 validation

Reply:
it's not clear to me why this is needed either. For a correct request, this is sufficient to pick out the address.

This tool isn't designed to validate requests. It's designed to find requests that have been validated by reviewers.

I guess I can read "need" as "please add ...".

Reply:
we're operating on the value - so this would allow an `agoric1"/><whatever"breakgoogleparsingwhoknows}`, which we could just not have any exposure to with validation.  Since there's manual review currently you could rely on that, but if it were me I would prefer exact control over the size and shape of thing for input validation's sake

Reply:
operating on the value is good justification. thanks.